server/datastore/mysql/policies.go

```diff
@@ -59,14 +59,17 @@ func (ds *Datastore) Policy(ctx context.Context, id uint) (*fleet.Policy, error)
 func (ds *Datastore) PolicyByName(ctx context.Context, name string) (*fleet.Policy, error) {
 	var policy fleet.Policy
 	err := sqlx.GetContext(ctx, ds.reader(ctx), &policy,
-		fmt.Sprint(`SELECT `+policyCols+`,
-		COALESCE(u.name, '<deleted>') AS author_name,
-		COALESCE(u.email, '') AS author_email,
-		(select count(*) from policy_membership where policy_id=p.id and passes=true) as passing_host_count,
-		(select count(*) from policy_membership where policy_id=p.id and passes=false) as failing_host_count
+		fmt.Sprint(`
+		SELECT `+policyCols+`,
+			COALESCE(u.name, '<deleted>') AS author_name,
+			COALESCE(u.email, '') AS author_email,
+			COALESCE(ps.passing_host_count, 0) AS passing_host_count,
+			COALESCE(ps.failing_host_count, 0) AS failing_host_count
 		FROM policies p
 		LEFT JOIN users u ON p.author_id = u.id
-		WHERE p.name=?`), name)
+		LEFT JOIN policy_stats ps ON p.id = ps.policy_id
+		WHERE p.name=?
```

Comment:
Should you do `AND ((p.team_id IS NULL AND ps.inherited_team_id = 0) OR (p.team_id IS NOT NULL))`? 

(From https://github.com/fleetdm/fleet/pull/14040/files#r1336100835.) 

In other words:
- If this is a global policy then you want to use the row that has ALL counts (inherited_team_id = 0)
- If this is a team policy, then there can only be one row (or zero) to use in this table.

Let me know if I'm making sense.
Same on the query below (`policyDB`)?